driver/src/main/java/oracle/nosql/driver/http/Client.java

```diff
@@ -1245,6 +1248,9 @@ Result processOKResponse(ByteInputStream in, Request kvRequest,
                 Serializer ser = kvRequest.createDeserializer(factory);
                 if (kvRequest instanceof QueryRequest ||
                     kvRequest instanceof PrepareRequest) {
+                    prepareResponseTestHook(kvRequest, in,
```

Comment:
Is the hook intended to be modifiable per-request or just per-handle session? If the latter can you just do a one-time call to know if there's a hook (i.e. property) set during construction to make the check more efficient? I'm not fond the amount of unconditional code path, even if it's small